### PR TITLE
feat(gesture): add support for blurring active inputs on gesture start

### PR DESCRIPTION
**Pull request by liamdebeasi**
blocks https://github.com/ionic-team/ionic-docs/pull/1331

## Pull request checklist

Please check if your PR fulfills the following requirements:
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build (`npm run build`) was run locally and any changes were pushed
- [x] Lint (`npm run lint`) has passed locally and any fixes were made for failures

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

Issue Number: https://github.com/ionic-team/ionic/issues/20588

## What is the new behavior?

- Menu gets and blurs any active element when swiping to open menu

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by liamdebeasi** [comment]
This may end up becoming a feature for the gesture util, but I'm not quite sure yet. Going to close for now until I can collect more feedback.

File: core/src/utils/gesture/index.ts [hunk 1]
```diff
@@ -141,7 +142,20 @@ export const createGesture = (config: GestureConfig): Gesture => {
     return true;
   };
 
+  const blurActiveElement = () => {
+    /* tslint:disable-next-line */
+    if (typeof document !== 'undefined') {
```
Comment (Haeven) [inline core/src/utils/gesture/index.ts]: Is there a reason you decided against a more general truthy check?
e.g `if (document) {`

Comment (liamdebeasi) [reply core/src/utils/gesture/index.ts]: `if (typeof document !== 'undefined') {` is a check to see if the script is being run inside a web browser or not. It allows us to support server-side rendering in Ionic Framework.

Comment (Haeven) [reply core/src/utils/gesture/index.ts]: I see! That makes sense, thank you for elaborating!

Comment (liamdebeasi) [inline core/src/utils/gesture/index.ts]: If used in a gesture, not sure if this is needed -- I don't think SSR would trigger this

Comment (liamdebeasi) [reply core/src/utils/gesture/index.ts:147]: talked with adam, going to keep it in - better safe than sorry

**Pull request merged**